scripts/lib/CIME/case/case_run.py

```diff
@@ -248,7 +248,7 @@ def _do_external(script_name, caseroot, rundir, lid, prefix):
     filename = "{}.external.log.{}".format(prefix, lid)
     outfile = os.path.join(rundir, filename)
     append_status("Starting script {}".format(script_name), "CaseStatus")
-    run_sub_or_cmd(script_name, [caseroot], (os.path.basename(script_name).split('.',1))[0], [caseroot], logfile=outfile)
+    run_sub_or_cmd(script_name, [caseroot], (os.path.basename(script_name).split('.',1))[0], [caseroot], logfile=outfile) # For sub, use case?
```

Comment:
wouldn't it just call them as externals until the interface was updated?

Reply:
You mean as shell commands? That would depend on if they are python programs or not.

Reply:
Pretty sure. Different number of arguments would be an AttributeError. But, since python is dynamically typed, it will be happy to pass along an argument of a type that the receiving function is not expecting.

Reply:
Sorry, just looking at this now.

Can you explain the purpose of this comment? Is this a reminder of something that needs to be changed in the future? If so, adding a little more to this comment so it's understandable by others would help... also, consider adding a TODO in the comment to call it out as such.

Reply:
I'm not sure we want to do it or not, just something to think about. It could be hard to track down all the scripts users are using for pre/post processing.